zcash_transparent/src/builder.rs

```diff
@@ -378,4 +451,365 @@ impl Bundle<Unauthorized> {
             authorization: Authorized,
         })
     }
+
+    /// Prepares the bundle for staged application of external signatures.
+    ///
+    /// This method computes the sighash for each transparent input using the provided
+    /// `calculate_sighash` closure.
+    #[cfg(feature = "transparent-inputs")]
+    pub fn prepare_transparent_signatures<F>(
+        self,
+        calculate_sighash: F,
+        secp_ctx: &secp256k1::Secp256k1<secp256k1::VerifyOnly>,
+    ) -> Result<TransparentSignatureContext<secp256k1::VerifyOnly>, Error>
+    where
+        F: Fn(SignableInput) -> [u8; 32], // The closure's signature
+    {
+        // Compute the sighashes for all inputs using the provided closure.
+        let sighashes = self
+            .authorization
+            .inputs
+            .iter()
+            .enumerate()
+            .map(|(index, info)| {
+                calculate_sighash(SignableInput {
+                    hash_type: SighashType::ALL,
+                    index,
+                    // for p2pkh, script_code is always the same as script_pubkey
+                    script_code: &info.coin.script_pubkey,
+                    script_pubkey: &info.coin.script_pubkey,
+                    value: info.coin.value,
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let num_inputs = self.authorization.inputs.len();
+        Ok(TransparentSignatureContext {
+            original_vin_unauthorized: self.vin,
+            original_vout: self.vout,
+            authorization_inputs: self.authorization.inputs,
+            sighashes, // The newly computed and owned Vec of sighashes
```

Comment:
```suggestion
            sighashes,
```
Comment is too obvious:
* it was computed just above;
* it's not usually necessary to comment on ownership except in particularly tricky cases, which this is not.